homeassistant/components/vevor_heater/sensor.py

```diff
@@ -0,0 +1,157 @@
+        key="target_power_level",
+        name="Target Power Level",
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    "current_power_level": SensorEntityDescription(
+        key="current_power_level",
+        name="Current Power Level",
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    "combustion_temperature": SensorEntityDescription(
+        key="combustion_temperature",
+        name="Combustion Chamber Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    "room_temperature": SensorEntityDescription(
+        key="room_temperature",
+        name="Room Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    "error:": SensorEntityDescription(
+        key="error",
+        name="Error",
+        device_class=SensorDeviceClass.ENUM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Vevor BLE heater sensors."""
+
+    coordinator: VevorHeaterUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    _LOGGER.debug("got sensors: %s", coordinator.data)
+
+    async_add_entities(
+        [VevorHeaterSensor(coordinator, sensor) for sensor in SENSORS_MAPPINGS.values()]
+    )
+
+
+class VevorHeaterSensor(CoordinatorEntity[VevorHeaterUpdateCoordinator], SensorEntity):
+    """Vevor BLE heater sensors for the device."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: VevorHeaterUpdateCoordinator,
+        entity_description: SensorEntityDescription,
+    ) -> None:
+        """Populate the entity with relevant data."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+
+        self._attr_unique_id = (
+            f"{coordinator.get_device_address()}_{entity_description.key}"
+        )
+        self._attr_device_info = DeviceInfo(
+            name=coordinator.data.name,
+            manufacturer="Vevor",
+            model="Vevor Heater",
+            connections={(CONNECTION_BLUETOOTH, coordinator.get_device_address())},
+        )
+        self._propagate_value()
+
+    @callback
+    def _propagate_value(self) -> None:
+        """Update attrs from device."""
+        if self.coordinator.data.status is not None:
+            self._attr_native_value = getattr(
+                self.coordinator.data.status, self.entity_description.key
+            )
```

Comment:
Instead of this use a small function for each description to get the value so it can be typed

Reply:
That works.  I think the current practice is to call it `value_fn`

Reply:
Perfect, renamed.